pom.xml

```diff
@@ -51,7 +51,6 @@
   <modules>
     <module>drools-benchmarks</module>
     <module>jbpm-benchmarks</module>
-    <module>optaplanner-benchmarks</module>
```

Comment:
Ok, sure

Reply:
I did that so to not harm others in master branch
I do not need the parent module and I thought that if the module is independent of others why to bother for example jbpm if OptaPlanner tests has compile error.

e624da07

Reply:
People _need_ to be bothered if something doesn't compile. Ideally, as soon as possible (that's why we have CI) so the compilation error doesn't even get merged.

Reply:
So now that you have returned the module, `optaplanner-benchmarks` is being built during the GithHub Actions check, which is good.

Reply:
Thanks, I was not aware of that best practice

Reply:
Resolved.

Reply:
There is another side effect, if we build from the root with the current setup then maven downloads dependencies for 7.x and 8.x they can clash so I start to think that this is a bad idea.
In that case I can resolve it like this:
I can create completlely side branch for OptaPlaner benchmarks and remove drools and jbpm tests from there. (They are currently not going to migrate this tests to the Kogito versions at all) WDYT?

Reply:
I don't think they clash. The dependency tree looks OK:
```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.kie:optaplanner-perf-benchmark:jar:1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-shade-plugin is missing. @ line 54, column 15
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] OptaPlanner Benchmarks Parent                                      [pom]
[INFO] OptaPlanner Performance Framework                                  [jar]
[INFO] OptaPlanner Performance Benchmark                                  [jar]
[INFO] 
[INFO] -------------------< org.kie:optaplanner-benchmarks >-------------------
[INFO] Building OptaPlanner Benchmarks Parent 1.0-SNAPSHOT                [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ optaplanner-benchmarks ---
[INFO] org.kie:optaplanner-benchmarks:pom:1.0-SNAPSHOT
[INFO] 
[INFO] -----------------< org.kie:optaplanner-perf-framework >-----------------
[INFO] Building OptaPlanner Performance Framework 1.0-SNAPSHOT            [2/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ optaplanner-perf-framework ---
[INFO] org.kie:optaplanner-perf-framework:jar:1.0-SNAPSHOT
[INFO] +- org.optaplanner:optaplanner-core:jar:8.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-api:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-internal:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-drools-model:jar:1.1.0.Final:compile
[INFO] |  |  \- org.drools:drools-model-compiler:jar:7.48.0.Final:compile
[INFO] |  |     +- org.drools:drools-canonical-model:jar:7.48.0.Final:compile
[INFO] |  |     +- com.github.javaparser:javaparser-core:jar:3.13.10:compile
[INFO] |  |     +- org.drools:drools-mvel-parser:jar:7.48.0.Final:compile
[INFO] |  |     \- org.drools:drools-mvel-compiler:jar:7.48.0.Final:compile
[INFO] |  +- org.kie.kogito:drools-core:jar:1.1.0.Final:compile
[INFO] |  |  +- org.drools:drools-core:jar:7.48.0.Final:compile
[INFO] |  |  |  \- org.kie.soup:kie-soup-xstream:jar:7.48.0.Final:compile
[INFO] |  |  +- org.kie.kogito:kogito-services:jar:1.1.0.Final:compile
[INFO] |  |  \- org.kie.kogito:drools-core-static:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:drools-compiler:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-compiler:jar:7.48.0.Final:runtime
[INFO] |  |     +- org.kie:kie-memory-compiler:jar:7.48.0.Final:compile
[INFO] |  |     +- org.drools:drools-ecj:jar:7.48.0.Final:runtime
[INFO] |  |     +- org.kie.soup:kie-soup-maven-support:jar:7.48.0.Final:runtime
[INFO] |  |     \- org.antlr:antlr-runtime:jar:3.5.2:runtime
[INFO] |  +- org.kie.kogito:drools-core-dynamic:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-core-reflective:jar:7.48.0.Final:compile
[INFO] |  +- org.apache.commons:commons-math3:jar:3.4.1:compile
[INFO] |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO] |  |  \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
[INFO] |  \- org.glassfish.jaxb:jaxb-runtime:jar:2.3.3:runtime
[INFO] |     +- org.glassfish.jaxb:txw2:jar:2.3.3:runtime
[INFO] |     +- com.sun.istack:istack-commons-runtime:jar:3.0.11:runtime
[INFO] |     \- com.sun.activation:jakarta.activation:jar:1.2.2:runtime
[INFO] +- org.optaplanner:optaplanner-examples:jar:8.1.0.Final:compile
[INFO] |  +- org.optaplanner:optaplanner-persistence-common:jar:8.1.0.Final:compile
[INFO] |  +- org.optaplanner:optaplanner-benchmark:jar:8.1.0.Final:compile
[INFO] |  |  +- org.optaplanner:optaplanner-persistence-jaxb:jar:8.1.0.Final:compile
[INFO] |  |  \- org.freemarker:freemarker:jar:2.3.28:compile
[INFO] |  +- org.kie.kogito:drools-decisiontables:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-decisiontables:jar:7.48.0.Final:runtime
[INFO] |  |     \- org.drools:drools-templates:jar:7.48.0.Final:runtime
[INFO] |  |        +- org.drools:drools-mvel:jar:7.48.0.Final:runtime
[INFO] |  |        |  \- org.kie.soup:kie-soup-project-datamodel-commons:jar:7.48.0.Final:runtime
[INFO] |  |        |     +- org.kie.soup:kie-soup-commons:jar:7.48.0.Final:runtime
[INFO] |  |        |     \- org.kie.soup:kie-soup-project-datamodel-api:jar:7.48.0.Final:runtime
[INFO] |  |        \- org.mvel:mvel2:jar:2.4.11.Final:runtime
[INFO] |  +- commons-io:commons-io:jar:2.6:compile
[INFO] |  +- org.glassfish:jakarta.json:jar:1.1.6:compile
[INFO] |  +- org.jfree:jfreechart:jar:1.5.0:compile
[INFO] |  +- org.jdom:jdom:jar:1.1.3:compile
[INFO] |  +- org.apache.poi:poi:jar:4.1.2:compile
[INFO] |  |  +- commons-codec:commons-codec:jar:1.13:compile
[INFO] |  |  +- org.apache.commons:commons-collections4:jar:4.4:compile
[INFO] |  |  \- com.zaxxer:SparseBitSet:jar:1.2:compile
[INFO] |  \- org.apache.poi:poi-ooxml:jar:4.1.2:compile
[INFO] |     +- org.apache.poi:poi-ooxml-schemas:jar:4.1.2:compile
[INFO] |     |  \- org.apache.xmlbeans:xmlbeans:jar:3.1.0:compile
[INFO] |     +- org.apache.commons:commons-compress:jar:1.19:compile
[INFO] |     \- com.github.virtuald:curvesapi:jar:1.06:compile
[INFO] +- org.optaplanner:optaplanner-persistence-xstream:jar:8.1.0.Final:compile
[INFO] +- org.apache.commons:commons-lang3:jar:3.9:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.26:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.2.3:runtime
[INFO] |  \- ch.qos.logback:logback-core:jar:1.2.3:runtime
[INFO] +- com.thoughtworks.xstream:xstream:jar:1.4.14:compile
[INFO] |  +- xmlpull:xmlpull:jar:1.1.3.1:compile
[INFO] |  \- xpp3:xpp3_min:jar:1.1.4c:compile
[INFO] \- org.openjdk.jmh:jmh-core:jar:1.26:compile
[INFO]    \- net.sf.jopt-simple:jopt-simple:jar:4.6:compile
[INFO] 
[INFO] -----------------< org.kie:optaplanner-perf-benchmark >-----------------
[INFO] Building OptaPlanner Performance Benchmark 1.0-SNAPSHOT            [3/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ optaplanner-perf-benchmark ---
[INFO] org.kie:optaplanner-perf-benchmark:jar:1.0-SNAPSHOT
[INFO] +- org.kie:optaplanner-perf-framework:jar:1.0-SNAPSHOT:compile
[INFO] |  +- org.optaplanner:optaplanner-persistence-xstream:jar:8.1.0.Final:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.9:compile
[INFO] |  \- com.thoughtworks.xstream:xstream:jar:1.4.14:compile
[INFO] |     +- xmlpull:xmlpull:jar:1.1.3.1:compile
[INFO] |     \- xpp3:xpp3_min:jar:1.1.4c:compile
[INFO] +- org.optaplanner:optaplanner-core:jar:8.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-api:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-internal:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:kogito-drools-model:jar:1.1.0.Final:compile
[INFO] |  |  \- org.drools:drools-model-compiler:jar:7.48.0.Final:compile
[INFO] |  |     +- org.drools:drools-canonical-model:jar:7.48.0.Final:compile
[INFO] |  |     +- com.github.javaparser:javaparser-core:jar:3.13.10:compile
[INFO] |  |     +- org.drools:drools-mvel-parser:jar:7.48.0.Final:compile
[INFO] |  |     \- org.drools:drools-mvel-compiler:jar:7.48.0.Final:compile
[INFO] |  +- org.kie.kogito:drools-core:jar:1.1.0.Final:compile
[INFO] |  |  +- org.drools:drools-core:jar:7.48.0.Final:compile
[INFO] |  |  |  \- org.kie.soup:kie-soup-xstream:jar:7.48.0.Final:compile
[INFO] |  |  +- org.kie.kogito:kogito-services:jar:1.1.0.Final:compile
[INFO] |  |  \- org.kie.kogito:drools-core-static:jar:1.1.0.Final:compile
[INFO] |  +- org.kie.kogito:drools-compiler:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-compiler:jar:7.48.0.Final:runtime
[INFO] |  |     +- org.kie:kie-memory-compiler:jar:7.48.0.Final:compile
[INFO] |  |     +- org.drools:drools-ecj:jar:7.48.0.Final:runtime
[INFO] |  |     +- org.kie.soup:kie-soup-maven-support:jar:7.48.0.Final:runtime
[INFO] |  |     \- org.antlr:antlr-runtime:jar:3.5.2:runtime
[INFO] |  +- org.kie.kogito:drools-core-dynamic:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-core-reflective:jar:7.48.0.Final:compile
[INFO] |  +- org.apache.commons:commons-math3:jar:3.4.1:compile
[INFO] |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
[INFO] |  |  \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
[INFO] |  \- org.glassfish.jaxb:jaxb-runtime:jar:2.3.3:runtime
[INFO] |     +- org.glassfish.jaxb:txw2:jar:2.3.3:runtime
[INFO] |     +- com.sun.istack:istack-commons-runtime:jar:3.0.11:runtime
[INFO] |     \- com.sun.activation:jakarta.activation:jar:1.2.2:runtime
[INFO] +- org.optaplanner:optaplanner-examples:jar:8.1.0.Final:compile
[INFO] |  +- org.optaplanner:optaplanner-persistence-common:jar:8.1.0.Final:compile
[INFO] |  +- org.optaplanner:optaplanner-benchmark:jar:8.1.0.Final:compile
[INFO] |  |  +- org.optaplanner:optaplanner-persistence-jaxb:jar:8.1.0.Final:compile
[INFO] |  |  \- org.freemarker:freemarker:jar:2.3.28:compile
[INFO] |  +- org.kie.kogito:drools-decisiontables:jar:1.1.0.Final:runtime
[INFO] |  |  \- org.drools:drools-decisiontables:jar:7.48.0.Final:runtime
[INFO] |  |     \- org.drools:drools-templates:jar:7.48.0.Final:runtime
[INFO] |  |        +- org.drools:drools-mvel:jar:7.48.0.Final:runtime
[INFO] |  |        |  \- org.kie.soup:kie-soup-project-datamodel-commons:jar:7.48.0.Final:runtime
[INFO] |  |        |     +- org.kie.soup:kie-soup-commons:jar:7.48.0.Final:runtime
[INFO] |  |        |     \- org.kie.soup:kie-soup-project-datamodel-api:jar:7.48.0.Final:runtime
[INFO] |  |        \- org.mvel:mvel2:jar:2.4.11.Final:runtime
[INFO] |  +- commons-io:commons-io:jar:2.6:compile
[INFO] |  +- org.glassfish:jakarta.json:jar:1.1.6:compile
[INFO] |  +- org.jfree:jfreechart:jar:1.5.0:compile
[INFO] |  +- org.jdom:jdom:jar:1.1.3:compile
[INFO] |  +- org.apache.poi:poi:jar:4.1.2:compile
[INFO] |  |  +- commons-codec:commons-codec:jar:1.13:compile
[INFO] |  |  +- org.apache.commons:commons-collections4:jar:4.4:compile
[INFO] |  |  \- com.zaxxer:SparseBitSet:jar:1.2:compile
[INFO] |  \- org.apache.poi:poi-ooxml:jar:4.1.2:compile
[INFO] |     +- org.apache.poi:poi-ooxml-schemas:jar:4.1.2:compile
[INFO] |     |  \- org.apache.xmlbeans:xmlbeans:jar:3.1.0:compile
[INFO] |     +- org.apache.commons:commons-compress:jar:1.19:compile
[INFO] |     \- com.github.virtuald:curvesapi:jar:1.06:compile
[INFO] +- org.openjdk.jmh:jmh-core:jar:1.26:compile
[INFO] |  \- net.sf.jopt-simple:jopt-simple:jar:4.6:compile
[INFO] +- org.openjdk.jmh:jmh-generator-annprocess:jar:1.26:provided
[INFO] +- org.slf4j:slf4j-api:jar:1.7.26:compile
[INFO] \- ch.qos.logback:logback-classic:jar:1.2.3:runtime
[INFO]    \- ch.qos.logback:logback-core:jar:1.2.3:runtime
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for OptaPlanner Benchmarks Parent 1.0-SNAPSHOT:
[INFO] 
[INFO] OptaPlanner Benchmarks Parent ...................... SUCCESS [  0.609 s]
[INFO] OptaPlanner Performance Framework .................. SUCCESS [  0.393 s]
[INFO] OptaPlanner Performance Benchmark .................. SUCCESS [  0.049 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.322 s
[INFO] Finished at: 2021-07-01T12:21:16+02:00
[INFO] ------------------------------------------------------------------------
```

Reply:
That is true, but if you use the idea it will update your dependency by default and downloads both to your local repository(

Reply:
But that is because of the other benchmarks (drools, jbpm), right? Whether `optaplanner-benchmarks` is module of `kie-benchmarks` or not has no effect on that. If you want to prevent IDEA from downloading 7.57.0-SNAPSHOT dependencies, you can open just `optaplanner-benchmarks`.

Reply:
Thanks for the hint, afraid that wont help much smb who is not aware of this(

Reply:
34532098